sphinx/application.py

```diff
@@ -203,12 +205,10 @@ def __init__(self, srcdir: str | os.PathLike[str], confdir: str | os.PathLike[st
         else:
             self._warning = warning
         self._warncount = 0
-        self.keep_going = warningiserror and keep_going
-        if self.keep_going:
-            self.warningiserror = False
-        else:
-            self.warningiserror = warningiserror
+        self.keep_going = bool(warningiserror)
```

Comment:
Since we do not early exit anymore, `keep_going` is always True.

Reply:
This is bad naming. Currently when ``keep_going`` is true, ``self.warningiserror`` is always false, and ``warningiserror`` is saved into ``self.keep_going``. So this preserves current behaviour.